src/datahike_postgres/core.clj

```diff
@@ -1,30 +1,55 @@
 (ns datahike-postgres.core
-  (:require [datahike.store :refer [empty-store delete-store connect-store scheme->index]]
+  (:require [datahike.store :refer [empty-store delete-store connect-store scheme->index default-config config-spec]]
+            [datahike.config :refer [int-from-env bool-from-env]]
             [hitchhiker.tree.bootstrap.konserve :as kons]
             [konserve-pg.core :as kp]
+            [environ.core :refer [env]]
+            [clojure.spec.alpha :as s]
             [superv.async :refer [<?? S]]))
 
-(defn convert-config
-  "convert datahike config to clojure.java.jdbc config"
-  [{:keys [username password host port dbname ssl sslfactory]}]
-  {:dbtype "postgresql"
-   :user username
-   :password password
-   :host host
-   :port port
-   :dbname dbname
-   :ssl ssl
-   :sslfactory sslfactory})
-
 (defmethod empty-store :pg [config]
   (kons/add-hitchhiker-tree-handlers
-   (<?? S (kp/new-pg-store (convert-config config)))))
+   (<?? S (kp/new-pg-store config))))
 
 (defmethod delete-store :pg [config]
-  (kp/delete-store (convert-config config)))
+  (kp/delete-store config))
 
 (defmethod connect-store :pg [config]
-  (<?? S (kp/new-pg-store (convert-config config))))
+  (<?? S (kp/new-pg-store config)))
 
 (defmethod scheme->index :pg [_]
   :datahike.index/hitchhiker-tree)
+
+(defmethod default-config :pg [config]
+  (merge
+   {:dbtype "postgresql"
+    :user (:datahike-store-user env)
+    :password (:datahike-store-password env)
+    :host (:datahike-store-host env)
+    :port (int-from-env :datahike-store-port nil)
+    :dbname (:datahike-store-dbname env)
+    :ssl (bool-from-env (:datahike-store-ssl env) nil)
+    :sslfactory (:datahike-store-ssl-factory env)}
+   config))
+
+(s/def :datahike.store.pg/backend #{:pg})
+(s/def :datahike.store.pg/user string?)
+(s/def :datahike.store.pg/password string?)
+(s/def :datahike.store.pg/host string?)
+(s/def :datahike.store.pg/dbtype string?)
+(s/def :datahike.store.pg/port int?)
+(s/def :datahike.store.pg/dbame string?)
```

Comment:
typo dbame -> dbname...funnily does not throw an error here

Reply:
but it would be `database` anyway when we stick to postgresql-jdbc-naming. I am not sure if there is a standard for it but it seems `database` is used widely.

Reply:
I'll follow these [specifications](http://clojure-doc.org/articles/ecosystem/java_jdbc/home.html#database-support).

Reply:
alright